apps/dashboard/app/models/batch_connect/session_context.rb

```diff
@@ -59,6 +59,37 @@ def method_missing(method_name, *arguments, &block)
     # @return [Boolean]
     def respond_to_missing?(method_name, include_private = false)
       /^(?<id>[^=]+)=?$/ =~ method_name.to_s && self[id] || super
+   end
+    
+
+    # @return [Boolean]
+    def attribute_cache_enabled?
+      self.any? {|v| v.opts[:cacheable]  } 
+    end
+    
+    
+    # @return [Boolean]   
+    def app_specific_cache_enabled?
+      self[:cacheable]
```

Comment:
I agree with the approach but not the implementation. The app specific cache could be set to true, set to false, or not set at all. These three different states have a different impact.

For example, assume the global cache enabled by default. If app specific cache is not set i.e. `nil`, we should use the cache. If app specific cache is set to `false`, we should not use the cache. Having a single boolean check `app_specific_cache_enabled?` doesn't differentiate between the `nil` case and the `false` case.